src/ProductsXmlFeed.php

```diff
@@ -436,23 +436,35 @@ private static function get_property_g_additional_image_link( $product, $propert
 	 */
 	private static function get_property_g_shipping( $product, $property ) {
 		$currency      = get_woocommerce_currency();
-		$entries       = array();
 		$shipping      = self::get_shipping();
 		$shipping_info = $shipping->prepare_shipping_info( $product );
 
 		if ( empty( $shipping_info ) ) {
 			return '';
 		}
 
+		$shipping_nodes = array();
+
 		/*
-		 * Entry is a comma separated string with values in the following format:
-		 *   COUNTRY:STATE:POST_CODE:SHIPPING_COST
+		 * Entry is a one or multiple XML nodes in the following format:
+		 *  <g:shipping>
+		 *		<g:country>...</g:country>
+		 *		<g:region>...</g:region>
+		 *		<g:service>...</g:service>
+		 *		<g:price>...</g:price>
+		 *	</g:shipping>
 		 */
 		foreach ( $shipping_info as $info ) {
-			$entries[] = "$info[country]:$info[state]:$info[name]:$info[cost] $currency";
+			$shipping_nodes[] =
+				'<g:shipping>' . PHP_EOL .
+					"\t\t\t\t<g:country>$info[country]</g:country>" . PHP_EOL .
+					( $info['state'] ? "\t\t\t\t<g:region>$info[state]</g:region>" . PHP_EOL : '' ) .
+					"\t\t\t\t<g:service>$info[name]</g:service>" . PHP_EOL .
+					"\t\t\t\t<g:price>$info[cost] $currency</g:price>" . PHP_EOL .
+				"\t\t\t</g:shipping>" . PHP_EOL;
```

Comment:
I think we can remove the last PHP_EOL as it is adding an extra empty line between the last g:shipping tag and the next tag:

![image](https://user-images.githubusercontent.com/2488994/154935124-806d42e8-97ae-4fbf-86de-f655ea08d3da.png)

Reply:
Done.